src/setuptools_scm/config.py

```diff
@@ -0,0 +1,49 @@
+""" configuration """
+from __future__ import print_function, unicode_literals
+import re
+import warnings
+
+from .utils import singleton
+
+DEFAULT_TAG_REGEX = r'^(?P<prefix>\w+-)?(?P<version>v?\d+(?:\.\d+){0,2}[^\+]+)(?P<suffix>\+.*)?$'
```

Comment:
the strict requirement for prefix/suffix as groups makes the parsing needlessly complex for other users,

Reply:
Easy to drop. I did that to have those parts available for reconstruction and because you have an explicit warning for dropping the suffix part which would need to go otherwise.
Shall I remove prefix and suffix then?

Reply:
the explicit warnigns are still needed, its just not clear what to do with the groups and regex requirements off hand

Reply:
Just found that it's (of course) possible to look before and after a group match to get prefix and suffix. Ergo those groups can be dropped entirely without loosing their information.